--- a/parser/BUILD.bazel
+++ b/parser/BUILD.bazel
@@ -1,0 +2,8 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+cc_library(
+    name = "headers",
+    hdrs = glob(["include/**/*.h"]),
+    textual_hdrs = glob(["include/**/*.c"]),
+    includes = ["include", "include/postgres"],
+)
@@ -93,0 +102 @@
+    cdeps = [":headers"],
@@ -96 +105,6 @@
-        "-Iparser/include -Iparser/include/postgres -g -fstack-protector -std=gnu99 -Wno-unknown-warning-option -Wno-typedef-redefinition -DXXH_NAMESPACE=PG_QUERY_",
+        "-g",
+        "-fstack-protector",
+        "-std=gnu99",
+        "-Wno-unknown-warning-option",
+        "-Wno-typedef-redefinition",
+        "-DXXH_NAMESPACE=PG_QUERY_",
